### ISSUE TITLE
Re-implementation to apply BinaryMoS to llama2-7b model

### ISSUE THREAD

**Issue by kimwin2** [dongwonjo/BinaryMoS#1]
Thank you for your efforts in releasing the code for the Binary LLM model. 
I would like to try out the process of training using BinaryMoS among your researches because PPL is impressive at 1bit Llama2-7B. 
How many A100 GPUs did you use to achieve a PPL of 7.88? And what were the epochs then? What learning rate did you use?

Thank you for your reply in advance.

**Comment by dongwonjo** [comment]
Thank you for your interest in our BinaryMoS.

For training the 1-bit LLaMA2-7B model with BinaryMoS, we used 2 A100 GPUs and trained for 3 epochs. The learning rate we used was 2E-5. Let me know if you have any further questions.

**Comment by kimwin2** [comment]
Thank you for your response first. I also wanted to use binarymos to perform QAT with 4 A100 GPUs, but it seems that it will take about 4-5 days due to the large dataset. Is this correct? If so, I think that if I utilize a method like bitdistiller, the QAT training will be completed within a day. Have you considered such a method?

**Comment by dongwonjo** [comment]
Thank you for your follow-up question.

When training with 2 A100 GPUs, the process took approximately 90 hours. Given that we utilize model parallelism, considering device communication overhead, it is entirely possible for the training to take around 4-5 days, as you mentioned.

Regarding BitDistiller, we have not considered that approach. Currently, we employ a straightforward KD method. However, incorporating a more advanced distillation technique like BitDistiller could be a promising direction for further improving training efficiency.

**Comment by kimwin2** [comment]
Thank you for sharing the detailed learning configuration. We are currently doing QAT using binarymos with LLM-QAT. Can you share the learning graph or the final (3epoch) logit loss and l2l loss values that you have achieved? If you can share the learning graph, it would be a great help. We are not sure if the model we are training is being learned properly.

We also tried to train the model with Bitdistiller, but failed. I guess Bitdistiller is using group quantization, which means the number 
 of scaling factors is much more then the binary mos's.
